lab_2_tokenize_by_bpe/main.py

```diff
@@ -58,6 +106,29 @@ def train(
     :param num_merges: required number of new tokens
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if (not isinstance(word_frequencies, dict) and word_frequencies is not None) or not isinstance(num_merges, int):
+        return None
+    while num_merges > 0:
+        dict_of_token_pairs = count_tokens_pairs(word_frequencies)
+        if dict_of_token_pairs is None:
+            return None
+        num_merges = min(num_merges, len(dict_of_token_pairs))
+        temp_list_of_max_token_pairs = []
+        list_of_max_token_pairs = []
+        max_token_pair = max(dict_of_token_pairs.values())
+        for token_pair in dict_of_token_pairs:
+            if dict_of_token_pairs[token_pair] == max_token_pair:
+                temp_list_of_max_token_pairs.append(token_pair)
+        max_token_pair_length = max(len(str(token_pair)) for token_pair in temp_list_of_max_token_pairs)
+        for token_pair in temp_list_of_max_token_pairs:
+            if len(str(token_pair)) == max_token_pair_length:
+                list_of_max_token_pairs.append(token_pair)
+        sorted_list_of_max_token_pairs = sorted(list_of_max_token_pairs)
```

Comment:
consider sorting by all those criteria via `key` argument instead of performing a lengthy filtering